lib/fh_cluster.js

```diff
@@ -49,21 +134,27 @@ function setOnWorkerListeningHandler() {
     var host = address.address || address.addressType === 4 ? '0.0.0.0' : '::';
     var addr = host + ':' + address.port;
     console.log('Cluster worker',  worker.id, 'is now listening at', addr);
+
+    //A new process is listening, need to check if any bound tasks need to be assigned.
+    boundTasks.setUpBoundTasks(singleWorkerTasks, cluster);
+  });
+
+  cluster.on('exit', function(){
+    //If a worker exits, check that all bound workers are on a running worker.
+    boundTasks.setUpBoundTasks(singleWorkerTasks, cluster);
   });
 }
 
 function setupOnExitBackOff(strategy, backoffResetTimeout) {
-  cluster.on('disconnect', function(worker) {
-    if (cluster.isMaster) {
-      var nextRetry = strategy.next();
-      console.log('Worker #', worker.id, 'disconnected. Will retry in',
-                  nextRetry, 'ms');
-
-      setTimeout(function() {
-        cluster.fork();
-        resetBackoffResetTimeout(strategy, backoffResetTimeout);
-      }, nextRetry);
-    }
+  cluster.on('exit', function(worker) {
```

Comment:
https://nodejs.org/docs/latest-v0.10.x/api/cluster.html#cluster_event_disconnect

It seems that the "exit" event is the final event that is emitted when a worker dies for any reason.

After the disconnect event, there seems to be some cleanup that happens with closing connections ect. It made sense to wait until this is done before transferring control to another worker.

(uncaughtException, kill -9 etc)

It seems to be the catchall event that is emitted.